internal/servers/worker/tcp_proxy.go

```diff
@@ -65,13 +65,16 @@ func (w *Worker) handleTcpProxyV1(connCtx context.Context, clientAddr *net.TCPAd
 	go func() {
 		defer connWg.Done()
 		_, err := io.Copy(netConn, tcpRemoteConn)
+		netConn.Close()
+		tcpRemoteConn.Close()
 		w.logger.Debug("copy from client to endpoint done", "error", err)
 	}()
 	go func() {
 		defer connWg.Done()
 		_, err := io.Copy(tcpRemoteConn, netConn)
+		tcpRemoteConn.Close()
+		netConn.Close()
 		w.logger.Debug("copy from endpoint to client done", "error", err)
```

Comment:
nit: It would be easier to scan logs for a clear error message, rather than having to parse them for an occurrence of a non `nil` error logline.
I am aware this is not a change you made in this specific PR so feel free to ignore it for now...

Reply:
Yeah I thought about gating this on `nil`. But from a debug perspective I thought it might be useful to keep as is. We can always change later.